packages/machine/test/integration/virtual-app-set-state-commitment.spec.ts

```diff
@@ -0,0 +1,165 @@
+    value => {
+      throw value;
+    },
+    reason => reason
+  );
+
+beforeAll(async () => {
+  [{}, wallet, networkId] = await connectToGanache();
+
+  const relevantArtifacts = [
+    { contractName: "AppRegistry", ...AppRegistry },
+    { contractName: "ETHBucket", ...ETHBucket },
+    { contractName: "StateChannelTransaction", ...StateChannelTransaction }
+  ];
+
+  network = {
+    // Fetches the values from build artifacts of the contracts needed
+    // for this test and sets the ones we don't care about to 0x0
+    ETHBalanceRefund: AddressZero,
+    ...relevantArtifacts.reduce(
+      (accumulator: { [x: string]: string }, artifact: WaffleLegacyOutput) => ({
+        ...accumulator,
+        [artifact.contractName as string]: artifact.networks![networkId].address
+      }),
+      {}
+    )
+  } as NetworkContext;
+
+  appRegistry = new Contract(
+    (AppRegistry as WaffleLegacyOutput).networks![networkId].address,
+    AppRegistry.abi,
+    wallet
+  );
+});
+
+beforeEach(() => {
+  signingKeys = getSortedRandomSigningKeys(2);
+
+  const users = signingKeys.map(x => x.address);
+
+  const stateChannel = StateChannel.setupChannel(
+    network.ETHBucket,
+    AddressZero,
+    users
+  ).setFreeBalance(AssetType.ETH, {
+    [users[0]]: WeiPerEther,
+    [users[1]]: WeiPerEther
+  });
+
+  const freeBalanceETH = stateChannel.getFreeBalanceFor(AssetType.ETH);
+
+  targetAppInstance = new AppInstance(
+    AddressZero,
+    users,
+    10,
+    freeBalanceETH.appInterface,
+    freeBalanceETH.terms,
+    true,
+    5,
+    0,
+    freeBalanceETH.toJson().latestState,
+    freeBalanceETH.toJson().latestNonce,
+    freeBalanceETH.timeout
+  );
+
+  c0 = new VirtualAppSetStateCommitment(
+    network,
+    targetAppInstance.identity,
+    65536,
+    targetAppInstance.timeout,
+    undefined,
+    undefined
+  );
+  s0 = signingKeys[0].signDigest(c0.hashToSign(true));
+});
+
+describe("The virtualAppSetState transaction generated by the commitment", () => {
+  jest.setTimeout(JEST_TEST_WAIT_TIME);
+
+  it("succeeds", async () => {
+    const c1 = new VirtualAppSetStateCommitment(
+      network,
+      targetAppInstance.identity,
+      65536,
+      targetAppInstance.timeout,
+      targetAppInstance.hashOfLatestState,
+      targetAppInstance.nonce
+    );
+    const s1 = signingKeys[1].signDigest(c1.hashToSign(false));
+
+    const txn = c1.transaction([s1], s0);
+
+    await wallet.sendTransaction({
+      ...txn,
+      gasLimit: SETSTATE_COMMITMENT_GAS
+    });
+
+    const contractAppState = await appRegistry.appStates(
+      targetAppInstance.identityHash
+    );
+
+    expect(contractAppState.appStateHash).toBe(
+      targetAppInstance.hashOfLatestState
+    );
+  });
+  it("fails with nonce above expiry", async () => {
+    const c1 = new VirtualAppSetStateCommitment(
+      network,
+      targetAppInstance.identity,
+      65536,
+      targetAppInstance.timeout,
+      targetAppInstance.hashOfLatestState,
+      65537
```

Comment:
There are several magic numbers in here, can you make them more clear w.r.t. intent